### ISSUE TITLE
Add inspection to detect duplicate keys in Cargo.toml

### ISSUE THREAD

**Issue by afetisov** [intellij-rust/intellij-rust#9453]
Having duplicate keys is invalid. However, currently there is no inspection to detect it.

It's an actually painful error to fix, because `cargo` stops at the first error, rather than provide a list of all invalid keys.

**Comment by Undin** [comment]
It seems that duplicate keys are invalid in TOML language in general. If so, the inspection should be implemented in the TOML plugin itself

**Comment by afetisov** [comment]
Yes, that makes sense. But I can't make a PR against the current repository of the plugin. What's the process for a feature request there?

**Comment by Undin** [comment]
Since TOML plugin was moved to intellij repository, now it's an issue in https://youtrack.jetbrains.com/issues/IDEA with `Land.TOML` subsystem

<img width="335" alt="Screenshot 2022-10-07 at 18 47 20" src="https://user-images.githubusercontent.com/2539310/194605672-da1282b5-033c-44ff-b880-c764f662eab1.png">

**Comment by avrong** [comment]
It's also was a part of https://github.com/intellij-rust/intellij-rust/issues/7180, most of which can be resolved with json schemas, and using it's parts such as path calculation and tracking for duplicates in this case

**Comment by Undin** [comment]
@avrong actually, duplicates detection based on some intermediate representation of TOML shouldn't be treated as a part of TOML schema support. It's a separate task (I think more low-level) not related to TOML schema support, but this intermediate representation can be used for schema support as well
